src/main/storm/mesos/MesosNimbus.java

```diff
@@ -890,7 +890,7 @@ private Credential getCredential(FrameworkInfo.Builder finfo) {
       if (StringUtils.isNotEmpty(secretFilename)) {
         try {
           // The secret cannot have a NewLine after it
-          credentialBuilder.setSecret(ByteString.readFrom(new FileInputStream(secretFilename)));
```

Comment:
Thanks @erikdw for pointing out and thanks @DarinJ for pushing the changing. I will get rid of this change.